src/libraries/System.Numerics.Tensors/tests/TensorTests.cs

```diff
@@ -2419,5 +2419,18 @@ public void TestIReadOnlyTMembers(TensorConstructor constructor)
             int expectedIndexValue = constructor.IsReversedStride ? 4 : 2;
             Assert.Equal(expectedIndexValue, tensorList[1]);
         }
+
+        [Theory]
+        [MemberData(nameof(GetConstructedTensors))]
+        public void TestGetEnumerator(Tensor<int> tensor)
+        {
+            static IEnumerable<int> GetExpected(Tensor<int> tensor)
+            {
+                for (int index = 0; index < tensor.Length; ++index)
+                    yield return tensor.GetValue(index);
+            }
+
+            Assert.Equal(GetExpected(tensor), tensor);
+        }
```

Comment:
Other things generally worth testing are disposing the enumerator multiple times, and reseting and using the enumerator again if reset is implemented.

Reply:
I've added tests for idempotent dispose and proper reset behavior. 

Is it okay for dispose to be a no-op, and for the enumerator to not throw when used after being disposed? I noticed most other struct enumerators in dotnet/runtime are implemented that way, I assume for performance.

Reply:
> Is it okay for dispose to be a no-op, and for the enumerator to not throw when used after being disposed? 

Yes.  You can see, for example, that the enumerators generated by the C# compiler often nop Dispose:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4oDchhGAzOVqQMKkDehp35jFS5HPQA8ASwB2AFwB8pAKLiArgFsYUAIaSYANXUAbRTADOACgCUXHpwI9bpAGbRSJiZNKjSAXlLEq70sKkOL7uANShZoI45ADs7jQ2PAC+hElAA==